src/react/ISV/components/BucketField.tsx

```diff
@@ -0,0 +1,230 @@
+import { Input } from '@scality/core-ui/dist/next';
+import {
+  FieldErrors,
+  useFieldArray,
+  useFormContext,
+  UseFormRegister,
+} from 'react-hook-form';
+import { useTheme } from 'styled-components';
+
+import { XCORE_NOT_AVAILABLE } from '../../next-architecture/ui/XCoreLibraryProvider';
+import { useXCoreLibrary } from '../../next-architecture/ui/XCoreLibraryProvider';
+import {
+  CapacityFormWithXcore,
+  CapacityFormSection,
+} from './ISVCapacityFormSection';
+import { unitChoices } from '../constants';
+
+const defaultBucketNameTooltip = (
+  <Text>Choose an unique name for your bucket</Text>
+);
+
+type BucketFieldProps = {
+  bucketNameTooltip?: React.JSX.Element;
+  platform?: string;
+};
+
+type BucketField = {
+  name: string;
+  tag?: string;
+  capacity?: string;
+  capacityUnit?: string;
+};
+
+type FormValues = {
+  buckets: BucketField[];
+  bucketNumber?: number;
+};
+
+const BucketNameFormGroup = ({
+  index,
+  errors,
+  bucketNamePlaceholder,
+  bucketNumber,
+  bucketNameTooltip,
+  register,
+}: {
+  index: number;
+  errors: FieldErrors<FormValues>;
+  bucketNamePlaceholder: string;
+  bucketNumber: number;
+  bucketNameTooltip: React.JSX.Element;
+  register: UseFormRegister<FormValues>;
+}) => (
+  <FormGroup
+    id={`bucketName-${index}`}
+    label={bucketNumber > 1 ? `Bucket #${index + 1} name` : 'Bucket name'}
+    required
+    labelHelpTooltip={bucketNameTooltip}
+    error={(errors?.buckets?.[index]?.name?.message as string) ?? ''}
+    helpErrorPosition="bottom"
+    content={
+      <Input
+        id={`bucketName-${index}`}
+        type="text"
+        autoComplete="off"
+        placeholder={bucketNamePlaceholder}
+        {...register(`buckets.${index}.name`)}
+      />
+    }
+  />
+);
+
+const BucketField = (fieldOverrides: BucketFieldProps) => {
+  const { bucketNameTooltip = defaultBucketNameTooltip, platform } =
+    fieldOverrides;
+
+  const theme = useTheme();
+  const [bucketNumber, setBucketNumber] = useState<number>(1);
+
+  const {
+    register,
+    control,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+
+  const { fields, append, remove, replace } = useFieldArray({
+    name: 'buckets',
+    control,
+  });
+
+  useEffect(() => {
+    if (fields.length === 0) {
+      append({
+        name: '',
+        tag: platform,
+        capacity: '0',
+        capacityUnit: unitChoices.TiB.toString(),
+      });
+    } else {
+      const updatedFields = fields.map((field) => ({
+        name: field.name,
+        tag: platform,
+        capacity: field.capacity,
+        capacityUnit: field.capacityUnit,
+      }));
+      replace(updatedFields);
+    }
+  }, [platform, fields.length]);
+
+  const handleBucketNumberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newNumber = e.target.valueAsNumber;
+
+    if (newNumber < 1 || newNumber > 20 || isNaN(newNumber)) {
+      return;
+    }
+
+    setBucketNumber(newNumber);
```

Comment:
I feel like bucketNumber state should be derived from the buckets length in the formhook state